tests/test_adapter.py

```diff
@@ -0,0 +1,370 @@
+from collections.abc import Collection
+from contextlib import contextmanager
+
+# In Python ≤ 3.8 you cannot annotate with “collections.abc.Collection[Item]”,
+# so we need to import typing.Collection for annotation instead.
+from typing import Collection as CollectionType
+from typing import Optional
+
+import attrs
+import pytest
+from itemadapter import ItemAdapter
+
+from zyte_common_items import Item, Product, ZyteItemAdapter
+
+from .test_items import _PRODUCT_ALL_KWARGS, _PRODUCT_MIN_KWARGS
+
+
+@contextmanager
+def configured_adapter():
+    ItemAdapter.ADAPTER_CLASSES.appendleft(ZyteItemAdapter)
+    try:
+        yield
+    finally:
+        ItemAdapter.ADAPTER_CLASSES.popleft()
+
+
+def test_asdict_all_fields():
+    product = Product(**_PRODUCT_ALL_KWARGS)
+    with configured_adapter():
+        adapter = ItemAdapter(product)
+        actual_dict = adapter.asdict()
+    expected_dict = {
+        "additionalProperties": [dict(name="foo", value="bar")],
+        "aggregateRating": dict(
+            bestRating=5.0,
+            ratingValue=2.5,
+            reviewCount=123,
+        ),
+        "availability": "InStock",
+        "brand": dict(name="Ka-pow"),
+        "breadcrumbs": [
+            dict(name="Level 1", url="http://example.com/level1"),
+            dict(name="Level 2", url="http://example.com/level1/level2"),
+        ],
+        "canonicalUrl": "https://example.com/product22",
+        "color": "white",
+        "currency": "USD",
+        "currencyRaw": "$",
+        "description": "Full freshness all over the fridge\n5 Conversion Modes on demand\nSuper Cooling Plus™",
+        "descriptionHtml": (
+            "<article><p>Full freshness all over the fridge</p>"
+            "<p>5 Conversion Modes on demand</p>"
+            "<p>Super Cooling Plus&trade;</p></article>"
+        ),
+        "features": ["Easily store fragile products.", "Bluetooth connectivity."],
+        "gtin": [dict(type="foo", value="bar")],
+        "images": [
+            dict(url="http://example.com/image1.png"),
+        ],
+        "mainImage": dict(url="http://example.com/image1.png"),
+        "metadata": dict(
+            dateDownloaded="2022-12-31T13:01:54Z",
+            probability=1.0,
+        ),
+        "mpn": "HSC0424PP",
+        "name": "White two-door refrigerator",
+        "price": "9999.99",
+        "productId": "A123DK9823",
+        "regularPrice": "11999.99",
+        "size": "XL",
+        "sku": "A123DK9823",
+        "style": "polka dots",
+        "url": "https://example.com/?product=product22",
+        "variants": [{}],
```

Comment:
That's an interesting case! On a first sight, I'd expect it to be dropped from the output.